metrics/heapster_test.go

```diff
@@ -46,41 +45,16 @@ var emptyMetricSink = &metricsink.MetricSink{}
 var emptyNodeLister = &cache.StoreToNodeLister{}
 var emptyPodLister = &cache.StoreToPodLister{}
 
-var testInsecurePort = 8080
 var testSecurePort = 6443
 
-func TestRunServer(t *testing.T) {
-	serverIP := fmt.Sprintf("http://localhost:%d", testInsecurePort)
-
-	go func() {
-		opt := getServerOptions()
-		opt.InsecurePort = testInsecurePort
-
-		server, err := app.NewHeapsterApiServer(opt, emptyMetricSink, emptyNodeLister, emptyPodLister)
-		if err != nil {
-			t.Fatalf("Could not create the API server: %v", err)
-		}
-		if err := server.RunServer(); err != nil {
-			t.Fatalf("Error in bringing up the server: %v", err)
-		}
-	}()
-	if err := waitForApiserverUp(serverIP); err != nil {
-		t.Fatalf("%v", err)
-	}
-	testSwaggerSpec(t, serverIP)
-	testAPIGroupList(t, serverIP)
-	testAPIGroup(t, serverIP)
-	testAPIResourceList(t, serverIP)
-}
-
 func TestRunSecureServer(t *testing.T) {
 	serverIP := fmt.Sprintf("https://localhost:%d", testSecurePort)
 
 	go func() {
 		opt := getServerOptions()
-		opt.InsecurePort = 0
-		opt.SecurePort = testSecurePort
-		opt.CertDirectory = "/tmp"
+		opt.SecureServing.ServingOptions.BindPort = testSecurePort
+		opt.SecureServing.ServerCert.CertDirectory = "/tmp"
+		opt.DisableAuthForTesting = true
```

Comment:
> you're starting a secure server without security.....

Yeah, given a loopback identity which you have access to, you should be using that.  followup I guess since its preexisting.